lib/internal/modules/esm/loader.js

```diff
@@ -497,18 +499,22 @@ function getHooksProxy() {
  * Register a single loader programmatically.
  * @param {string} specifier
  * @param {string} [parentURL]
- * @returns {void}
+ * @param {any} [data] Arbitrary data passed to loader's `initialize` hook
+ * @param {any[]} transferList Objects in `data` that are changing ownership
+ * @returns {any}
  * @example
  * ```js
  * register('./myLoader.js');
  * register('ts-node/esm', import.meta.url);
  * register('./myLoader.js', import.meta.url);
  * register(new URL('./myLoader.js', import.meta.url));
+ * register('./myLoader.js', import.meta.url, {banana: 'tasty'});
+ * register('./myLoader.js', import.meta.url, someArrayBuffer, [someArrayBuffer]);
  * ```
  */
-function register(specifier, parentURL = 'data:') {
+function register(specifier, parentURL = 'data:', data = undefined, transferList = []) {
```

Comment:
Let's not create a throw away array and keep the `transferList` as undefined instead.
```suggestion
function register(specifier, parentURL = 'data:', data = undefined, transferList) {
```